### DISCUSSION TITLE
Named children in components

### DISCUSSION THREAD

**Discussion by tombuente** [a-h/templ#1004] [General]
Coming from other template engines it's often possible to have multiple slots. The closest equivalent I could find in is `children`.
Specifically, this example:
```html
templ showAll() {
	@wrapChildren() {
		<div>Inserted from the top</div>
	}
}

templ wrapChildren() {
	<div id="wrapper">
		{ children... }
	</div>
}
```
Imagine you have a `layout` component and a detail view for an `item`. For the detail view to work, additional JavaScript is required, you'd usually load it in `<head>`. But I don't see how to do that, as there are no named `children` slots. You'd need something like this:
```html
templ detail(item item) {
	@layout() {
		@children1 { <script src="https://..."></script>}
		@children2 { <div>Inserted from the top</div> }
	}
}

templ layout() {
	<head>
		{ children1... }
	</head>
	<div id="wrapper">
		{ children2... }
	</div>
}
```
Here, the `children1` slot would be used for inserting additional scripts in the `<head>`, while `children2` would contain content for the body of the layout. This structure allows more control over where different pieces of content are injected, similar to what other template engines might offer with named slots.

**Comment by pharrisee** [comment]
Funnily enough I just pointed you at some hints on the Go Discord channel.  Have a look at my playground repo, https://github.com/pharrisee/stove especialy the web folder for ideas on how things can be built as components.

I don't think it answers you completely but it might raise some ideas on how it might be done.

**Reply by pharrisee** [reply]
The other thing to say is that you can use templ.Components as regular go variables, so they can be passed in to the layout, e.g.:

```go
templ detail(item item) {
	@layout(headMatter()) {
		<div>Inserted from the top</div>
	}
}

templ detail2() {
    @layout(headMatter2()) { 
        <div>A second page with other requirements</div>
    }
}

templ headMatter() {
    <script src="https://some.js.lib.js"></script>
}

templ headMatter2() {
    <script src="https://another.js.lib.js"></script>
}



templ layout(headMatter templ.Component) {
	<head>
                <script src="/your/local.js"></script>
		@headMatter
	</head>
	<div id="wrapper">
		{ children... }
	</div>
}
```
